manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
```

Comment:
Anything is possible, we just need a base pattern we want to go with and then let those who wish to customize there after.